src/fit.jl

```diff
@@ -91,16 +91,23 @@ function train!(
 
     start_iter = get_iter_number(estimator) + 1
     end_iter = start_iter + num_iterations - 1
+    total_metrics_evals = sum((((start_iter:end_iter) .- 1) .% estimator.metric_freq) .== 0)
```

Comment:
We need to compute this (and some other things down the line) because of the way store scores works, it was getting OOB error with previous implemention because it assumed all trainings start from scratch which wasn't true since `update` was implemented for MLJ. This one tells us the size of the array to allocate in store scores